mono/mini/mini-generic-sharing.c

```diff
@@ -4059,6 +4059,29 @@ mini_is_gsharedvt_variable_signature (MonoMethodSignature *sig)
 	}
 	return FALSE;
 }
+
+MonoMethod*
+mini_method_to_shared (MonoMethod *method)
+{
+	if (!mono_method_is_generic_impl (method))
+		return NULL;
+
+	ERROR_DECL (error);
+
+	// This pattern is based on add_extra_method_with_depth.
+
+	if (mono_method_is_generic_sharable_full (method, TRUE, TRUE, FALSE))
+		// gshared over reference type
+		method = mini_get_shared_method_full (method, SHARE_MODE_NONE, error);
+	else if (mono_method_is_generic_sharable_full (method, FALSE, FALSE, TRUE))
+		// gshared over valuetype (or primitive?)
+		method = mini_get_shared_method_full (method, SHARE_MODE_GSHAREDVT, error);
+	else
+		return NULL;
```

Comment:
Arguably this should return method. All the callers want that. But this preserves more clearly information as to if it was shared (which can also be gleaned perhaps by comparing output to input).